File: server/bcephal-reco/src/main/java/com/moriset/bcephal/reconciliation/domain/PartialRecoItem.java
/**
 * 
 */
package com.moriset.bcephal.reconciliation.domain;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Data;

/**
 * @author Moriset
 *
 */
@Data
public class PartialRecoItem {

	private Long gridId;
	private Long id;
	private boolean left;
	private BigDecimal amount;
	private BigDecimal reconciliatedAmount;
	private BigDecimal remainningAmount;
	
	@JsonIgnore
	private boolean primary;
	
	@JsonIgnore
	private BigDecimal realAmount;
	
	@JsonIgnore
	public BigDecimal buildRealAmount(boolean useCreditDebit, boolean isDebit) {
		realAmount = remainningAmount;
		if(useCreditDebit && isDebit) {
			realAmount = BigDecimal.ZERO.subtract(remainningAmount);
		}
		return realAmount;
	}
	
}

File: client/bcephal-client-model/Reconciliation/PartialRecoItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Reconciliation
{
    public class PartialRecoItem
    {
        public long? Id { get; set; }
        public bool Left { get; set; }
        public decimal? Amount { get; set; }
        public decimal? ReconciliatedAmount { get; set; }
        public decimal? RemainningAmount { get; set; }

    }
}
